pkgs/development/python-modules/clickhouse-connect/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytestCheckHook
+  ### build_requires
+, cython_3
+  ### install_requires
+, certifi
+, importlib-metadata
+, urllib3
+, pytz
+, zstandard
+, lz4
+  ### extras_require
+, sqlalchemy
+, numpy
+, pandas
+, pyarrow
+, orjson
+  ### not in tests_require, but should be
+, pytest-dotenv
+}:
+buildPythonPackage rec {
+  pname = "clickhouse-connect";
+  version = "0.6.8";
+
+  format = "setuptools";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    repo = "clickhouse-connect";
+    owner = "ClickHouse";
+    rev = "v${version}";
+    hash = "sha256-T1vUXhOngPvA4wrg5jVzIiqj0B0gEnk/BS5OdopHgkA=";
+  };
+
+  nativeBuildInputs = [ cython_3 ];
+  setupPyBuildFlags = [ "--inplace" ];
+  enableParallelBuilding = true;
+
+  propagatedBuildInputs = [
+    certifi
+    importlib-metadata
+    urllib3
+    pytz
+    zstandard
+    lz4
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook pytest-dotenv ]
+    ++ passthru.optional-dependencies.sqlalchemy
+    ++ passthru.optional-dependencies.numpy;
+
+  # these tests require a running clickhouse instance
+  pytestFlagsArray = [
+    "--deselect=tests/integration_tests"
+    "--deselect=tests/tls"
+  ];
```

Comment:
Why not use disabledTestFiles? That also checks if the directory still exists

Reply:
Because I forget about all the ways there are to ignore tests 😂